### PR TITLE
Add constructor snippets

### PR DESCRIPTION
**Pull request by weeman1337**
- Refactoring like mentioned in the issue
- In addition to moving the the snippet classes to the `WorseReflection` namespace I changed the structure like the other formatters. Basically there is an `ObjectFormatter` capable of formatting all the things.
- Add constructor completion

Relates to phpactor/phpactor#1247

Extension PR phpactor/completion-worse-extension#16

### CODE REVIEW

File: tests/Integration/Bridge/TolerantParser/ReferenceFinder/NameSearcherCompletorTest.php [hunk 1]
```diff
@@ -27,51 +25,69 @@ public function testComplete(string $source, array $expected): void
...
         yield 'class' => [
-            '<?php class Foobar {} :int {}; new Foo<>', [
+            '<?php class Foobar { public function __construct(int $cparam) {} } :int {}; new Foo<>', [
                 [
-                    'type' => Suggestion::TYPE_CLASS,
-                    'name' => 'Foobar',
+                    'type'              => Suggestion::TYPE_CLASS,
+                    'name'              => 'Foobar',
                     'short_description' => 'Foobar',
+                    'snippet'           => 'Foobar(${1:\\$cparam})${0}',
```
Comment (dantleech) [inline tests/Integration/Bridge/TolerantParser/ReferenceFinder/NameSearcherCompletorTest.php:33]: as the CS fixer doesn't pick this up I won't block it, but in general there is no whitespace alignment in Phpactor.

**Comment by dantleech** [comment]
Thanks!

**Comment by dantleech** [comment]
There seems to be an issue with both this and the function completor:

image

At least on my machine, it's selecting the text too far back... I think this is the issue you mentioned r.e. auto import?

File: tests/Integration/Bridge/TolerantParser/ReferenceFinder/NameSearcherCompletorTest.php (hunk 1 above)
Comment (weeman1337) [reply tests/Integration/Bridge/TolerantParser/ReferenceFinder/NameSearcherCompletorTest.php:33]: Now I know. Maybe this should be added then to the CS Fixer.

**Comment by weeman1337** [comment]
> At least on my machine, it's selecting the text too far back... I think this is the issue you mentioned r.e. auto import? can confirm it only happens on import.

Yes exactly. That happens if I enable command execution.

**Comment by dantleech** [comment]
@weeman1337 I've disabled both this and the function snippets by default in Phpactor now, but it can be renabled with `completion_worse.experimental => true` in configuration.

One issue is when completing type hints:

```
public functio foo(Bar
```

gives 

```
function foo (Bar()
```

for example

**Comment by weeman1337** [comment]
That means at the place where creating the snippets I have to find out whether or not there is a precending `new`, correct? Do you have a clue about that?

**Comment by dantleech** [comment]
Not exactly. You have access to the `Node` for example [here](https://github.com/phpactor/completion/blob/475ce0b835526abd2975780676fc224cc4fb26a2/lib/Bridge/TolerantParser/ReferenceFinder/NameSearcherCompletor.php#L39) from which you can `$node->parent` or `$node->getFirstDescendantNode(/** type of new create node here **/)`

**Pull request merged**